crates/subspace-farmer/src/cluster/cache.rs

```diff
@@ -218,22 +223,39 @@ where
         })
         .collect::<Vec<_>>();
 
-    select! {
-        result = identify_responder(&nats_client, &caches_details, cache_group, identification_broadcast_interval).fuse() => {
-            result
-        },
-        result = write_piece_responder(&nats_client, &caches_details).fuse() => {
-            result
-        },
-        result = read_piece_index_responder(&nats_client, &caches_details).fuse() => {
-            result
-        },
-        result = read_piece_responder(&nats_client, &caches_details).fuse() => {
-            result
-        },
-        result = contents_responder(&nats_client, &caches_details).fuse() => {
-            result
-        },
+    if index == 0 {
```

Comment:
Why `index == 0` results in a different behavior?

Reply:
1. The boolean with a good name would be more transparent if we have no usage for `index` as an integer.
2. Do you think redesign by introducing a separate function with non-duplicated calls is applicable here? Those ifs with selects look rather awkward.

Reply:
I think boolean makes sense, will refactor. Thanks for suggestion!